detection-rules/impersonation_ms_teams_invite.yml

```
@@ -4,29 +4,45 @@ type: "rule"
 severity: "high"
 source: |
   type.inbound
-  and strings.icontains(body.current_thread.text, 'Microsoft Teams')
-  and strings.icontains(body.current_thread.text, 'join the meeting now')
-  and strings.contains(body.current_thread.text, 'Meeting ID:')
-  and strings.contains(body.current_thread.text, 'Passcode:')
-  
+  and (
+    (
+      strings.icontains(body.current_thread.text, 'Microsoft Teams')
+      and strings.icontains(body.current_thread.text, 'join the meeting now')
+      and strings.contains(body.current_thread.text, 'Meeting ID:')
+      and strings.contains(body.current_thread.text, 'Passcode:')
+    )
+    or 3 of (
+      strings.icontains(body.current_thread.text, "internal"),
+      strings.icontains(body.current_thread.text, "teams"),
+      strings.icontains(body.current_thread.text, "message"),
+      strings.icontains(body.current_thread.text, "meeting")
+    )
+  )
   // not a reply
   and length(headers.references) == 0
   and headers.in_reply_to is null
   // few links
   and length(body.links) < 10
-
+  
   // no unsubscribe links
   // common in newsletters which link to a webinar style event
   and not any(body.links, strings.icontains(.display_text, "unsub"))
-
-  // one of the links contains "join the meeting now"
-  and any(body.links, .display_text =~ "join the meeting now")
   
-  // the "join the meeting now" link does not go to microsoft
-  and all(filter(body.links, .display_text =~ "join the meeting now"),
-          .href_url.domain.root_domain not in ("microsoft.com", "microsoft.us")
-          // rewriters often abstract the link
-          and .href_url.domain.root_domain not in $bulk_mailer_url_root_domains
+  // one of the links contains is a CTA
+  and (
+    (
+      any(body.links, .display_text =~ "join the meeting now")
+      // the "join the meeting now" link does not go to microsoft
+      and all(filter(body.links, .display_text =~ "join the meeting now"),
+              .href_url.domain.root_domain not in (
+                "microsoft.com",
+                "microsoft.us"
+              )
+              // rewriters often abstract the link
+              and .href_url.domain.root_domain not in $bulk_mailer_url_root_domains
+      )
+    )
+    or any(body.links, strings.icontains(.display_text, "play recording"))
```

isntead of looping body.links again, add the "or" statement to the existing loop?
```suggestion
  // one of the links contains is a CTA
  and any(body.links, .display_text =~ "join the meeting now" or .display_text, "play recording")
      // the "join the meeting now" link does not go to microsoft
      and all(filter(body.links, .display_text =~ "join the meeting now"),
              .href_url.domain.root_domain not in (
                "microsoft.com",
                "microsoft.us"
              )
              // rewriters often abstract the link
              and .href_url.domain.root_domain not in $bulk_mailer_url_root_domains
      )
```